Framework/PythonInterface/plugins/algorithms/ReflectometrySliceEventWorkspace.py

```diff
@@ -78,7 +78,15 @@ def _slice_input_workspace(self):
         alg.setProperty("ExcludeSpecifiedLogs", False)
         alg.setProperty("TimeSeriesPropertyLogs", 'proton_charge')
         alg.execute()
-        return mtd[self._output_ws_group_name]
+        # Ensure the run number for the child workspaces is stored in the
+        # sample logs as a string (FilterEvents converts it to a double).
+        group = mtd[self._output_ws_group_name]
+        for ws in group:
+            if ws.run().hasProperty('run_number'):
+                run_number = int(ws.run()['run_number'].value)
+                AddSampleLog(Workspace=ws, LogName='run_number', LogType='String',
```

Comment:
Maybe avoid calling the algorithm `AddSampleLog` which has `InOut` property and ADS magic, maybe use
```
ws.run().addProperty('run_number', run_number)
```